src/components/Actions/GsuiteRoute.tsx

```diff
@@ -10,13 +11,9 @@ const GsuiteProtectedRoute = ({ Component, ...rest }: any) => {
     <Route
       {...rest}
       render={() => {
-        return auth_status.is_verified === true ? (
-          Component
-        ) : (
-          <Redirect exact to="/gsuite" />
```

Comment:
In this component you'll still need it to redirect to the `/gsuite` endpoint. Reason being is that Cloudflare Access only monitors that one endpoint to check if the person has signed in or not. Let's say a user navigates for the first time to an arbitrary protected route like `/admin`... Cloudflare Access will not trigger on that endpoint. We need to redirect them to the `/gsuite` endpoint, which will cause Cloudflare Access to run and authenticate them. Then we can navigate them back to `/admin` and let them continue on their way. If we render the `Authorize.tsx` component here it'll just enter an infinite loop of refreshing the page every couple of seconds.

Reply:
So instead of redirecting to the `Authorize` component, do we need to copy functionality of `Authorize` into `GsuiteProtectedRoute` and redirect to `/gsuite` instead?

Reply:
We should still redirect to the Authorize component. Think of it this way. In order to authenticate we need 2 things to happen. One we need to be on the `/gsuite` endpoint. Two, we refresh the page and after Cloudflare Access does its thing we need to copy the jwt into its respective atom. We redirect to the `/gsuite` endpoint for part 1 & the `Authorize.tsx` component takes care of the second point. 

In the `App.tsx` file we should have a `<Route path="/gsuite" component={Authorize}/>`.

Reply:
Need a separate route`<Route path="/gsuite" Component={<Authorize />} />`. A standard route is used on the `/gsuite` route since this is the specific route that cloudflare needs to trigger on. Then component Authorize to trigger cloudflare and initialize `jwt_gsuite` with the `CF_Authorize` cookie and verify the cookie. When a <GsuiteProtectedRoute />  is used for a specific route, if the `auth_status.is_verified: false`  the user is redirected to `/gsuite` route to go through process of authenticating then back to wherever they were going. If they were already verified then it just renders the component that was protected.